src/contracts/core/PaymentCoordinator.sol

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../interfaces/IPaymentCoordinator.sol";
+import "../libraries/Merkle.sol";
+import "@openzeppelin-upgrades/contracts/proxy/utils/Initializable.sol";
+import "@openzeppelin-upgrades/contracts/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "forge-std/Test.sol";
+
+
+
+/**
+ * @title Contract used to coordinate payments from AVSs to operators and in particular the subsequency splitting of earnings from operators to stakers
+ * @author Layr Labs, Inc.
+ */
+contract PaymentCoordinator is 
+    IPaymentCoordinator,
+    Initializable,
+    OwnableUpgradeable,
+    Test
+{
+    using SafeERC20 for IERC20;
+
+    /// @notice address approved to post new Merkle roots
+    address public rootPublisher;
+
+    uint256 public merkleRootActivationDelay;
+
+    /// @notice maximum BIPS
+    uint256 public constant MAX_BIPS = 10000;
+
+    /// @notice Array of roots of posted Merkle trees, as well as associated data like tree height
+    MerkleRootPost[] public merkleRoots;
+
+    /// @notice Mapping token => recipient => cumulative amount *claimed*
+    mapping(IERC20 => mapping(address => uint256)) public cumulativeTokenAmountClaimedByRecipient;
+
+    /// @notice Mapping token => cumulative amount earned by EigenLayer
+    mapping(IERC20 => uint256) public cumulativeEigenLayerTokeEarnings;
```

Comment:
typo: "Toke" => "Token"

Reply:
also what is this?

Reply:
to elaborate, we can't just store the cumulative earnings, we have to have some mechanism for these tokens to leave this contract

Reply:
oh, I see, it gets reset to 0 when `withdrawEigenlayerShare` is called. In that case it's not a 'cumulative' number, recommend removing this wording from the name.

Reply:
wait what's the difference in meaning - people in the office think they're the same?

Reply:
IMO:

- `cumulative` implies all-time (only increases)
- `accumulated` implies some kind of accumulation process (so in general it increases) but does not exclude the possibility of it decreasing.
- personally I don't think we should use either. if you need to attach a qualifier, `stored` might be more neutral